src/xlsx/mod.rs

```diff
@@ -1885,160 +1885,228 @@ where
     Ok(merge_cells)
 }
 
-struct CellReference {
-    row: u32,
-    col: u32,
-    absolute_row: bool,
-    absolute_col: bool,
+enum Reference {
+    Cell {
+        row: u32,
+        col: u32,
+        absolute_row: bool,
+        absolute_col: bool,
+    },
+    Row {
+        row: u32,
+        absolute: bool,
+    },
+    Column {
+        col: u32,
+        absolute: bool,
+    },
```

Comment:
The later logic might be simpler if this enum contained `RowRange` and `ColumnRange` variants.

Reply:
I think this might just shift the complexity. Individual row/column ranges aren't valid (e.g. 1 isn't valid, it has to be 1:1 afaik). We'd need to have the parser handle both parts at once which I think would be implemented as having an intermediate parser that returns this structure anyway which effectively is what's already there (the code parses each part of the range (`{before}:{after}`) and then handles the different cases.

I might not be thinking it through properly but lmk what you think.